salt/pillar/__init__.py

```diff
@@ -260,40 +272,24 @@ class Pillar(object):
     '''
     def __init__(self, opts, grains, minion_id, saltenv, ext=None, functions=None,
                  pillar=None, pillarenv=None):
+
+        self.orig_opts = opts
+        self._functions = functions
         self.minion_id = minion_id
         # Store the file_roots path so we can restore later. Issue 5449
-        self.actual_file_roots = opts['file_roots']
+        self.actual_file_roots = self.orig_opts['file_roots']
         # use the local file client
-        self.opts = self.__gen_opts(opts, grains, saltenv=saltenv, ext=ext, pillarenv=pillarenv)
-        self.client = salt.fileclient.get_file_client(self.opts, True)
+        self._opts = self.__gen_opts(self.orig_opts, grains, saltenv=saltenv, ext=ext, pillarenv=pillarenv)
 
-        if opts.get('file_client', '') == 'local':
-            opts['grains'] = grains
+        # FIXME: if we change self.orig_opts then it is no longer "original".  Hmmm.
+        if self.orig_opts.get('file_client') == 'local':
+            self.orig_opts['grains'] = grains
 
-        # if we didn't pass in functions, lets load them
```

Comment:
Are we handling this case any more? I'd like to understand why this was removed. It's not obvious to me.